src/polyswarm_api/__init__.py

```diff
@@ -371,19 +381,24 @@ def _fix_result(self, result):
         with open(to_scan, 'rb') as fobj:
             return await self.scan_fileobj(fobj, os.path.basename(to_scan))
 
-    async def search_hash(self, to_scan, hash_type='sha256'):
+    async def search_hash(self, hash_string):
         """
         Search for a single hash using the PS API asynchronously.
 
-        :param to_scan: Hash to search for
-        :param hash_type: Hash type [sha256|sha1|md5]
+        :param hash_string: Hash to search for
         :return: JSON report file
         """
         async with self.get_semaphore:
+
+            hash_type = self._get_hash_type(hash_string)
```

Comment:
Check the hash type before requesting the semaphore.

Reply:
done